R/build_schedule.R

```diff
@@ -8,23 +8,24 @@
 ##'
 ##' @title Build Cohort Schedule
 ##' @param p Parameters object
+##' @param ctrl Control object
 ##' @return A Parameters object, with schedule components set.  The
 ##' output offspring produced is also available as an attribute
 ##' \code{birth_rate}.
 ##' @author Rich FitzJohn
 ##' @export
-build_schedule <- function(p) {
+build_schedule <- function(p, env = make_environment(), ctrl = scm_base_control()) {
```

Comment:
How will `make_environment` know what type of environment to make? Same for `scm_base_control`. Would it work to change default to something like  `function(p, env = make_environment(p$type), ctrl = scm_base_control(p$type))`?